src/main/java/org/broadinstitute/hellbender/utils/pairhmm/VectorLoglessPairPDHMM.java

```diff
@@ -0,0 +1,399 @@
+    }
+    // /**
+    // * {@inheritDoc}
+    // */
+    //// @Override
+    // public void initialize(final int readMaxLength, final int haplotypeMaxLength
+    // ) {
+    // super.initialize(readMaxLength, haplotypeMaxLength);
+    //
+    // branchMatchMatrix = new
+    // double[paddedMaxReadLength][paddedMaxHaplotypeLength];
+    // branchInsertionMatrix = new
+    // double[paddedMaxReadLength][paddedMaxHaplotypeLength];
+    // branchDeletionMatrix = new
+    // double[paddedMaxReadLength][paddedMaxHaplotypeLength];
+    // // do not need to call super.initialize()
+    // int numHaplotypes = haplotypes.size();
+    // mHaplotypeDataArray = new HaplotypeDataHolder[numHaplotypes];
+    // int idx = 0;
+    // haplotypeToHaplotypeListIdxMap.clear();
+    // for (final Haplotype currHaplotype : haplotypes) {
+    // mHaplotypeDataArray[idx] = new HaplotypeDataHolder();
+    // mHaplotypeDataArray[idx].haplotypeBases = currHaplotype.getBases();
+    // haplotypeToHaplotypeListIdxMap.put(currHaplotype, idx);
+    // ++idx;
+    // }
+    // }
+
+    // /**
+    // * {@inheritDoc}
+    // */
+    // @Override
+    // public void initialize(final List<Haplotype> haplotypes, final Map<String,
+    // List<GATKRead>> perSampleReadList,
+    // final int readMaxLength, final int haplotypeMaxLength) {
+    //
+    // }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void computeLog10Likelihoods(final LikelihoodMatrix<GATKRead, PartiallyDeterminedHaplotype> logLikelihoods,
+                                        final List<GATKRead> processedReads,
+                                        final PairHMMInputScoreImputator inputScoreImputator,
+                                        final int rangeForReadOverlapToDeterminedBases) {
+
+        if (processedReads.isEmpty()) {
+            return;
+        }
+        if (doProfiling) {
+            startTime = System.nanoTime();
+        }
+        // (re)initialize the pairHMM only if necessary
+        final int readMaxLength = findMaxReadLength(processedReads);
+        final int haplotypeMaxLength = findMaxAlleleLength(logLikelihoods.alleles());
+        final int readCount = processedReads.size();
+        final List<PartiallyDeterminedHaplotype> alleles = logLikelihoods.alleles();
+        final int alleleCount = alleles.size();
+        final int totalComps = readCount * alleleCount;
+        // logger.info("Total Reads: " + readCount + "Total Alleles: " + alleleCount +
+        // "Total Comparisons: " + totalComps);
+        // totalCalls += 1;
+
+        ReadDataHolder[] readDataHolders = new ReadDataHolder[readCount];
+        for (int readIndex = 0; readIndex < readCount; readIndex++) {
+            GATKRead read = processedReads.get(readIndex);
+            final PairHMMInputScoreImputation inputScoreImputation = inputScoreImputator.impute(read);
+            readDataHolders[readIndex] = new ReadDataHolder();
```

Comment:
I think we should set all the values on ReadDataHolder before plugging it into the array, rather than looking it up in the array many times.  It would save some characters and maybe some cycles too depending on how well the optimizer handles this.

Reply:
Might want to change it in PairHmm too.